validator/client/runner.go

```diff
@@ -305,6 +306,33 @@ func runHealthCheckRoutine(ctx context.Context, v iface.Validator, eventsChan ch
 				return
 			}
 			isHealthy := tracker.CheckHealth(ctx)
+			if !isHealthy && len(hosts) > 1 && features.Get().EnableBeaconRESTApi {
+				for i, url := range hosts {
+					if url == v.RetrieveHost() {
+						next := (i + 1) % len(hosts)
+						log.Infof("Beacon node at %s is not responding, switching to %s", url, hosts[next])
+						v.UpdateHost(hosts[next])
+
+						km, err := v.Keymanager()
+						if err != nil {
+							log.WithError(err).Fatal("Could not get keymanager")
+						}
+						slot, err := v.CanonicalHeadSlot(ctx)
+						if err != nil {
+							log.WithError(err).Error("Could not get canonical head slot")
+							return
+						}
+						go func() {
+							// deadline set for 1 epoch from call to not overlap.
+							epochDeadline := v.SlotDeadline(slot + params.BeaconConfig().SlotsPerEpoch - 1)
```

Comment:
this deadline needs to change as we don't have a check if it's epoch start so this deadline will be wrong. I think it might be better to just have a default value.